core/node.go

```diff
@@ -143,9 +198,51 @@ func (n *Node) ID() peer.ID {
 // Start causes the Node to continuously send messages to and receive messages
 // from its peers. It blocks until an error is encountered or `Stop` is called.
 func (n *Node) Start() error {
+	// Note: dht.Bootstrap has a somewhat confusing name. It doesn't automatically
+	// connect to the bootstrap peers. It just starts the background process of
+	// searching for new peers.
+	if err := n.dht.Bootstrap(n.ctx); err != nil {
+		return err
+	}
+
+	// Advertise ourselves for the purposes of peer discovery.
+	discovery.Advertise(n.ctx, n.routingDiscovery, n.config.RendezvousString, discovery.TTL(advertiseTTL))
+
+	if n.config.UseBootstrapList {
+		if err := n.connectToBootstrapList(); err != nil {
+			return err
+		}
+	}
+
 	return n.mainLoop()
 }
 
+func (n *Node) connectToBootstrapList() error {
+	log.WithField("bootstrapPeers", dht.DefaultBootstrapPeers).Info("connecting to bootstrap peers")
+	connectCtx, cancel := context.WithTimeout(n.ctx, defaultNetworkTimeout)
```

Comment:
You probably want a lower timeout than 30 seconds for connections -- IIRC we tend to use 5 seconds.

Reply:
👍 In my experience, WebRTC connections can take longer, especially with third-party signaling. I'll reduce it for now and take another look once we plug in the WebRTC transport.